File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
  if (s.length !== t.length) return false;
  let objLetter = {};

  for (let c of s) {
    objLetter[c] = (objLetter[c] || 0) + 1;
  }

  for (let c of t) {
    if (!objLetter[c]) return false;
    objLetter[c]--;
  }

  return true;
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
        
        count_s = {}
        count_t = {}
        
        for letter in range(len(s)):
            count_s[s[letter]] = count_s.get(s[letter], 0) + 1
            count_t[t[letter]] = count_t.get(t[letter], 0) + 1
            
        return count_s == count_t
            
        